relay-server/src/actors/spooler/mod.rs

```diff
@@ -1,3 +1,30 @@
+//! This module contains the [`BufferService`], which responsible for spooling of the incoming
+//! envelopes, either in memory or on persistent storage.
+//!
+//! The main entry point for the [`BufferService`] is the [`Buffer`] interface. Which currently
+//! supports:
+//! - [`Enqueue`] - enqueueing message into the backend storage
+//! - [`DequeueMany`] - dequeueing all the requested [`QueueKey`] keys
+//! - [`RemoveMany`] - Removing and dropping the requested [`QueueKey`] keys.
+//! - [`Health`] - checking the health of the [`BufferService`]
+//!
+//! To make sure the [`BufferService`] is fast the responsive, especially in the normal working
+//! conditions, it keeps the internal [`BufferState`] state, which defines where the spooling will
+//! be happening.
```

Comment:
I'm not sure if I understand this paragraph. It seems to me it suggests that `BufferState`'s purpose is to make `BufferService` fast.